File: android-app/app/src/main/java/com/aiproj/mobile/data/models/Document.kt
package com.aiproj.mobile.data.models

import com.google.gson.annotations.SerializedName

/**
 * 文档数据模型
 */
data class Document(
    val id: Int,
    @SerializedName("task_id")
    val taskId: Int,
    val title: String,
    val content: String,
    val type: String? = "markdown",
    val status: String? = "draft",
    @SerializedName("created_at")
    val createdAt: String? = null,
    @SerializedName("updated_at")
    val updatedAt: String? = null,
    @SerializedName("created_by")
    val createdBy: Int? = null,
    @SerializedName("updated_by")
    val updatedBy: Int? = null,
    val version: Int = 1
)

/**
 * 文档类型枚举
 */
enum class DocumentType {
    MARKDOWN,
    TXT,
    PDF;

    companion object {
        fun fromString(value: String?): DocumentType {
            return when (value?.lowercase()) {
                "markdown", "md" -> MARKDOWN
                "txt", "text" -> TXT
                "pdf" -> PDF
                else -> MARKDOWN
            }
        }
    }
}

/**
 * 文档状态枚举
 */
enum class DocumentStatus {
    DRAFT,
    PUBLISHED,
    ARCHIVED;

    companion object {
        fun fromString(value: String?): DocumentStatus {
            return when (value?.lowercase()) {
                "draft" -> DRAFT
                "published" -> PUBLISHED
                "archived" -> ARCHIVED
                else -> DRAFT
            }
        }
    }
}

/**
 * 文档请求模型
 */
data class DocumentRequest(
    val title: String,
    val content: String,
    val type: String = "markdown",
    val status: String = "published"
)

/**
 * 文档列表响应
 */
data class DocumentListResponse(
    val success: Boolean,
    val message: String? = null,
    val data: List<Document>? = null,
    val error: String? = null
)

File: ios-backup-20251014/AI-Proj-iOS/Models/Document.swift
//
//  Document.swift
//  AI-Proj iOS
//
//  文档相关数据模型
//

import Foundation

/// 文档模型
struct Document: Identifiable, Codable, Hashable {
    let id: Int
    let title: String
    let content: String
    let projectId: Int?
    let taskId: Int?
    let ownerId: Int
    let ownerName: String?
    let createdAt: Date
    let updatedAt: Date
    let version: Int
    let status: DocumentStatus
    let type: DocumentType

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case content
        case projectId = "project_id"
        case taskId = "task_id"
        case ownerId = "owner_id"
        case ownerName = "owner_name"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case version
        case status
        case type
    }
}

/// 文档状态
enum DocumentStatus: String, Codable, CaseIterable {
    case draft = "draft"
    case published = "published"
    case archived = "archived"

    var displayName: String {
        switch self {
        case .draft: return "草稿"
        case .published: return "已发布"
        case .archived: return "已归档"
        }
    }
}

/// 文档类型
enum DocumentType: String, Codable, CaseIterable {
    case markdown = "markdown"
    case text = "txt"
    case pdf = "pdf"

    var displayName: String {
        switch self {
        case .markdown: return "Markdown"
        case .text: return "文本"
        case .pdf: return "PDF"
        }
    }
}

/// 文档响应模型
struct DocumentResponse: Codable {
    let success: Bool
    let data: Document?
    let message: String?
}

/// 文档列表响应模型
struct DocumentListResponse: Codable {
    let success: Bool
    let data: [Document]?
    let message: String?
}
